### PR TITLE
Extend open_files

### PR DESCRIPTION
**Pull request by tijko**
This allows for tracking of anonymous inodes and also adds a field for `mnt_id` inside of `popenfile`.  

> mnt_id This field, present since Linux 3.15, is the ID of the
> mount point containing this file.  See the description
> of /proc/[pid]/mountinfo

### CODE REVIEW

File: psutil/_pslinux.py [hunk 1]
```diff
@@ -159,8 +159,14 @@ class IOPriority(enum.IntEnum):
                 'read_time', 'write_time',
                 'read_merged_count', 'write_merged_count',
                 'busy_time'])
-popenfile = namedtuple(
-    'popenfile', ['path', 'fd', 'position', 'mode', 'flags'])
+
+popenfile = namedtuple('popenfile', ['path', 'fd', 'position', 'mode', 
+    'flags', 'mnt_id', 'anon_fields'])
+
```
Comment (tijko) [inline psutil/_pslinux.py]: Added the `mnt_id` field and a `anon_fields` which defaults to an empty `dict` if a regular file.

File: psutil/_pslinux.py [hunk 2]
```diff
@@ -159,8 +159,14 @@ class IOPriority(enum.IntEnum):
...
+ppoll_tfd = namedtuple('ppoll_tfd', ['events', 'data'])
+pinotify = namedtuple('pinotify_attrs', ['ino', 'sdev', 'mask', 'ignored_mask', 
+     'fhandle_bytes', 'fhandle_type', 'f_handle'])
+
```
Comment (tijko) [inline psutil/_pslinux.py]: Two new `namedtuple`s for `ppoll` and `inotify` descriptors.  Using the fd's as the keys and these `namedtuple`s as the values.  The other anon inodes `signalfd` and `eventfd`, don't contain multiple fields and can use just their event masks as the values.

**Comment by giampaolo** [comment]
What's the use case for this? What's an anonymous file? What's a `mnt_id`? You're introducing a difference in term of API for one platform, so it must be worth it.

**Comment by tijko** [comment]
The anonymous files are open files that don't have a corresponding inodes (i.e. they do not have a path on the filesystem).  Examples of these would be created using `epoll_create`, `inotify_init`, `signalfd`, `eventfd`, and others.  Nevertheless, there is still very useful data that is contained in their respective `/proc/$pid/fdinfo` entries.  For instance, if a process had numerous fd's being monitored by an `epoll` fd, you can then see which events were masked into said poll.  Another use (and the one that I had been needing) would be if you have an `inotify` fd in a process and needed to see what the inode of contained watch is, you would be able to find this under the `inotify`s `ino` field.

The `mnt_id` has been exposed since [Linux 3.15](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=49d063cb353265c3af701bab215ac438ca7df36d).  Which needed when a there are two file descriptors with the same path but, are on different mount points with separate namespaces.

**Comment by a-sk** [comment]
I would like to see `mountinfo` information for a given process.

**Comment by giampaolo** [comment]
Closing this out as outdated.

**Pull request closed without merging**